### ISSUE TITLE
gh-actions: No CI provider detected.

### ISSUE THREAD

**Issue by nschloe** [codecov/codecov-python#209]
E.g., https://github.com/nschloe/tikzplotlib/pull/329/checks?check_run_id=208746870.

**Comment by hugovk** [comment]
+1

Our options now:

1. Save the token at https://codecov.io/gh/org/repo/settings as `CODECOV_TOKEN` at https://github.com/org/repo/settings/secrets and add it to the workflow YAML:

```
      env:
        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Downside: not available to forks. And coverage is most needed from PRs from forks.

2. Save the token at https://codecov.io/gh/org/repo/settings as `CODECOV_TOKEN` at in plaintext in the workflow YAML:

```
      env:
        CODECOV_TOKEN: 12345678-abcd-1234-abcd-1234567890ab
```

Downside: shown to everyone. Perhaps not such a big deal, it's only useful for uploading coverage. And even https://codecov.io/gh/org/repo/settings suggests adding it to `codecov.yml`. Would be cleaner if forks can use their own tokens (or better yet, have it detected automatically :)

**Comment by pquentin** [comment]
GitHub Actions does not support this use case yet, see https://community.codecov.io/t/whitelist-github-action-servers-to-upload-without-a-token/491/10 for more details.

**Comment by nschloe** [comment]
The bash tool works for me, too. I make sure to run it on pushes to master and on pull requests,
```
on:
  push:
    branches:
      - master
  pull_request:
```
and submit it for only one of my matrix builds:
```
    - name: Submit to codecov
      if: ${{ matrix.python-version == '3.8' }}
      run: bash <(curl -s https://codecov.io/bash)
```

**Comment by thomasrockhu** [comment]
This was fixed in `2.1.6` (https://github.com/codecov/codecov-python/pull/275)
